### DISCUSSION TITLE
MessageBox

### DISCUSSION THREAD

**Discussion by NormskiJJO** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3483] [Ideas]
How about a standard material design MessageBox, having to rely on MaterialDeisgnExtention for AlertDialog and ConfirmationDialog.
It would be nice for a build in standard MessageBox.

**Comment by MichelMichels** [comment]
This seems out of scope to me. Could you elaborate some more on how you would use this and how this should look like?

My WPF projects always use MVVM architecture and as a side-effect I don't use MessageBox.

**Reply by Keboo** [reply]
Just to provide a bit more context. There is a potential desire to provide some additional dialog helpers. However, the API of a MessageBox is problematic in that it is a synchronous blocking call, it also lacks the ability to know what should be covered by the dialog. Any implementation we provide would probably resemble MahApps (which is a possible solution to use today), where the Window supports showing a dialog that then covers all of the Window's content. There are some outstanding questions of what to do if there are multiple instances of the same window. 

At present there are no plans to add this, but it is something that I would be happy to consider PRs, or API designs for.
